### DISCUSSION TITLE
“Mosh session finished!” immediately after connection attempt

### DISCUSSION THREAD

**Discussion by elliotec** [blinksh/blink#1582] [Support] [answered]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration

- Blink v15.0.7.536
- iPad OS version 15.4.1
- iPad Pro 12.9 Inch 5th Generation
- Mosh 1.3.2 on the host, UDP ports 60000 - 61000 open.
- SSH keys configured as per docs in blink config. SSH works perfectly fine.

### Describe the bug

I try to connect to my host with mosh and it fails silently with “Mosh session finished!” and no other output unless I do the `—verbose` flag.

Mosh is installed on the host at the location in the output. 

I’ve even copied the file at `~/.ssh/known_hosts` to that weird `/etc/ssh/ssh_known_hosts` location from the output. Trying to navigate to that weird location on the iPad of course just gives me auth errors.

The same output happens when I try to connect via the IP, but it prompts for a password at some point and still fails. 

`ssh elliotec` produces the expected result and connects to the host shell instantly. 
 
Thanks in advance for your help.

Expected behavior:
`mosh elliotec` Should connect me to the host shell via mosh.

Screenshot:

![7D9FBD78-BF82-4186-B8D2-A65F1AA4FBA9](https://user-images.githubusercontent.com/4829208/169675045-4a7fb709-38d3-4ff2-9e35-3e1325f77c4d.jpeg)

<details>
  <pre>
blink> mosh elliotec --verbose
MoshClient:DEBUG:"/usr/bin/mosh-server" new -s -c 256 -l LC_ALL=en_US.UTF-8
MoshClient:DEBUG:ssh -v -o compression=no -t elliotec -- "/usr/bin/mosh-server" new -s -c 256 -l LC_ALL=en_US.UTF-8 A socket_callback_connected: Socket connection callback: 1 (0)
ssh_key_cmp: key types don't match!
ssh_known_hosts_read_entries: Failed to open the known_hosts file '/etc/ssh/ssh_known_hosts': No such file or directory
ssh_key_cmp: key types don't match!
ssh_packet_userauth_failure: Access denied for 'none'. Authentication that can continue: publickey,password,keyboard-interactive
ssh_agent_get_ident_count: Answer type: 12, expected answer: 12
ssh_packet_userauth_failure: Access denied for 'publickey'. Authentication that can continue: publickey,password,keyboard-interactive
ssh_packet_global_request: Invalid SSH_MSG_GLOBAL_REQUEST packet
channel_request: Channel request env failed
channel_request: Channel request env failed 

Mosh session finished! </pre>
</details>

**Comment by elliotec** [comment]
This appears to be an issue caused by the fact I’m trying to mosh into WSL2. Windows apparently doesn’t support Mosh at all, so I’m not sure this is possible.

**Comment by carloscabanero** [comment, marked as answer]
Just my 2cents, as I never tried mosh under WSL. From the output, it looks like you are actually being connected, but the session finishes immediately. This may happen if a command on the remote immediately runs and exits (which seems weird in this case as you did not execute anything), or maybe something on the windows side (may be similar to https://github.com/mobile-shell/mosh/issues/781 ?).

**Comment by arusa** [comment]
I have the same problem while trying to connect to an Ubuntu server using Blink 14 from my iPad pro.

I haven’t used it for a while, but I think last time I tried it from the same iPad to the same Ubuntu server it was working.

**Comment by arusa** [comment]
Could be some bug with Blink:

https://www.reddit.com/r/BlinkShell/comments/ufp67s/issue_with_mosh_connections/

**Comment by arusa** [comment]
Oh no, sorry, I found the solution. It was a command in the mosh configuration for this particular server. When there is a command configured mosh executes it and in my case the command exited immediately
